azurerm/internal/services/databricks/databricks_workspace_resource_test.go

```diff
@@ -50,6 +46,84 @@ func TestAccDatabricksWorkspace_requiresImport(t *testing.T) {
 	})
 }
 
+func TestAccDatabricksWorkspace_machineLearning(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_workspace", "test")
+	r := DatabricksWorkspaceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.machineLearning(data, "standard"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+// This is currently untestable due to https://github.com/Azure/azure-rest-api-specs/issues/8075
+// Per jlichwa: It is as designed, purge protection is irreversible. Otherwise hacker would be able
+// to delete your key vault without option to recover. #please-close
+// Unable to delete key vault since purge protection is required by the resource:
+// Code="StorageAccountOperationFailed" Message="Unable to update storage account 'dbstoragel2bko37m6ebog': 'Keyvault policy recoverable is not set'."
+// If I Add Purge Protection to be true the clean up gets the below error:
+// keyvault.BaseClient#PurgeDeletedKey: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error.
+// Status=403 Code="Forbidden" Message="The user, group or application 'appid=XXXX;oid=XXXX;iss=https://sts.windows.net/XXXX/' does not have keys
+// purge permission on key vault 'acctest-kv-rczeo;location=westeurope'. InnerError={"code":"ForbiddenByPolicy"}
+// If you look at the azurerm_key_vault_access_policy that was created it does have purge enable, but since the default policy is 90 days you cannot purge
+// the keys for at least 90 days.
+
+// func TestAccDatabricksWorkspace_customerManagedKey(t *testing.T) {
+// 	data := acceptance.BuildTestData(t, "azurerm_databricks_workspace", "test")
+// 	r := DatabricksWorkspaceResource{}
+
+// 	data.ResourceTest(t, r, []acceptance.TestStep{
+// 		{
+// 			Config: r.customerManagedKeysStepOne(data, "premium"),
+// 			Check: acceptance.ComposeTestCheckFunc(
+// 				check.That(data.ResourceName).ExistsInAzure(r),
+// 			),
+// 		},
+// 		data.ImportStep(),
+// 		{
+// 			Config: r.customerManagedKeysStepTwo(data, "premium"),
+// 			Check: acceptance.ComposeTestCheckFunc(
+// 				check.That(data.ResourceName).ExistsInAzure(r),
+// 			),
+// 		},
+// 		data.ImportStep(),
+// 		{
+// 			Config: r.customerManagedKeysStepThree(data, "premium"),
```

Comment:
how come this is three steps?

Reply:
That is the way the API configured to work:
1) Enable CMK (this creates the MSI identity for the workspace)
2) Actually pass the BYOK key info so the DBFS can be encrypted by the workspace worker
3) Verify we can disable BYOK and revert to Microsoft Managed encryption